### PR TITLE
chore: take segents in memory and current seg while triggering reten…

### PR DESCRIPTION
**Pull request by savannahar68**
refactor retention logic

## What does this PR do?
- Summarize your change (**mandatory**)

## How does this PR work? (optional)
- Need a brief introduction for the changed logic (optional)

## Refer to a related PR or issue link
- (Related PR or issue)

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests. (for non-documentation changes)

### CODE REVIEW

File: coredb/src/lib.rs [hunk 1]
```diff
@@ -548,14 +548,21 @@ impl CoreDB {
       // TODO: this does not need to read all_segments_summaries from disk - can use the one already
```
Comment (vinaykakade) [inline coredb/src/lib.rs:548]: This TODO should be removed. No need to read all_segments_summaries from disk

File: coredb/src/lib.rs [hunk 2]
```diff
@@ -548,14 +548,21 @@ impl CoreDB {
...
       // in memory in Index::all_segments_summaries()
       let all_segments_summaries = index_entry.1.read_all_segments_summaries().await?;
+      let memory_segments = index_entry.1.get_memory_segments_numbers();
+      let current_segment_number = index_entry.1.get_current_segment_number();
+
+      // create set of segment to skip from retention which is combination of memory segment and current_segment_number
```
Comment (vinaykakade) [inline coredb/src/lib.rs:554]: nit: which is combination of memory segment and current_segment_number -> which is union of the segments in memory and the current_segment_number

File: coredb/src/lib.rs [hunk 3]
```diff
@@ -548,14 +548,21 @@ impl CoreDB {
       // TODO: this does not need to read all_segments_summaries from disk - can use the one already
       // in memory in Index::all_segments_summaries()
       let all_segments_summaries = index_entry.1.read_all_segments_summaries().await?;
+      let memory_segments = index_entry.1.get_memory_segments_numbers();
```
Comment (vinaykakade) [inline coredb/src/lib.rs:551]: nit: rename to memory_segment_numbers? (As these are segment numbers - not the actual segments or segment references)

**Comment by vinaykakade** [comment]
Pl fix typo in the PR description

**Review by vinaykakade — APPROVED** [review]

File: coredb/src/lib.rs (hunk 1 above)
Comment (savannahar68) [reply coredb/src/lib.rs:548]: done

File: coredb/src/lib.rs (hunk 2 above)
Comment (savannahar68) [reply coredb/src/lib.rs:554]: updated comment

File: coredb/src/lib.rs (hunk 3 above)
Comment (savannahar68) [reply coredb/src/lib.rs:551]: updated comment

**Pull request merged**